rdbms/src/main/java/org/spine3/server/storage/jdbc/entity/JdbcRecordStorage.java

```diff
@@ -21,87 +21,124 @@
  * The implementation of the entity storage based on the RDBMS.
  *
  * @param <I> the type of entity IDs
- * @see JdbcStorageFactory
  * @author Alexander Litus
+ * @see JdbcStorageFactory
  */
 public class JdbcRecordStorage<I> extends RecordStorage<I> {
 
     private final DataSourceWrapper dataSource;
 
-    private final EntityStorageQueryFactory<I> queryFactory;
-
-    private final Descriptors.Descriptor stateDescriptor;
+    private final RecordStorageQueryFactory<I> queryFactory;
 
     /**
      * Creates a new storage instance.
      *
-     * @param dataSource            the dataSource wrapper
-     * @param multitenant           defines is this storage multitenant
-     * @param queryFactory          factory that generates queries for interaction with entity table
-     * @throws DatabaseException    if an error occurs during an interaction with the DB
+     * @param dataSource   the data source wrapper
+     * @param multitenant  defines if this storage is multitenant or not
+     * @param queryFactory factory that generates queries for interaction with entity table
+     * @throws DatabaseException if an error occurs during an interaction with the DB
      */
     public static <I> JdbcRecordStorage<I> newInstance(DataSourceWrapper dataSource,
                                                        boolean multitenant,
-                                                       EntityStorageQueryFactory<I> queryFactory, Descriptors.Descriptor stateDescriptor)
+                                                       RecordStorageQueryFactory<I> queryFactory)
             throws DatabaseException {
-        return new JdbcRecordStorage<>(dataSource, multitenant, queryFactory, stateDescriptor);
+        return new JdbcRecordStorage<>(dataSource, multitenant, queryFactory);
     }
 
-    protected JdbcRecordStorage(DataSourceWrapper dataSource, boolean multitenant, EntityStorageQueryFactory<I> queryFactory, Descriptors.Descriptor stateDescriptor)
+    protected JdbcRecordStorage(DataSourceWrapper dataSource, boolean multitenant,
+                                RecordStorageQueryFactory<I> queryFactory)
             throws DatabaseException {
         super(multitenant);
         this.dataSource = dataSource;
         this.queryFactory = queryFactory;
-        this.stateDescriptor = stateDescriptor;
         queryFactory.setLogger(LogSingleton.INSTANCE.value);
-        queryFactory.newCreateEntityTableQuery().execute();
+        queryFactory.newCreateEntityTableQuery()
+                    .execute();
+    }
+
+    @SuppressWarnings("ProhibitedExceptionThrown") // NPE by the contract
+    @Override
+    public void markArchived(I id) {
+        checkNotNull(id);
+        final boolean recordExists = queryFactory.newMarkArchivedQuery(id)
+                                                 .execute();
+        if (!recordExists) {
+            // The NPE is required by the contract of the method
+            throw new NullPointerException(
```

Comment:
I think you should try investigating how to suppress this particular warning for the PMD rule.

Maybe you'll discover that's a Codacy bug.